src/common/components/clusterConfiguration/OpenShiftVersionSelect.tsx

```diff
@@ -37,15 +38,18 @@ const OpenShiftVersionSelect: React.FC<OpenShiftVersionSelectProps> = ({ version
     values: { highAvailabilityMode, openshiftVersion },
     setFieldValue,
   } = useFormikContext<ClusterCreateParams>();
-
+  const supportLevelData = React.useContext(FeatureSupportLevelContext);
   const selectedVersion = versions.find((v) => v.value === openshiftVersion) || versions[0];
   React.useEffect(() => {
     if (highAvailabilityMode === 'None' && !isSNOSupportedVersion(selectedVersion)) {
       const firstSupportedVersionValue = versions.find(isSNOSupportedVersion)?.value;
       setFieldValue('openshiftVersion', firstSupportedVersionValue);
     }
   }, [highAvailabilityMode, selectedVersion, setFieldValue, versions]);
-
+  React.useEffect(() => {
+    supportLevelData.openshiftVersion = openshiftVersion;
```

Comment:
why are you changing the context value directly ? I think it needs to be done via callback & state to be reactive

Reply:
This is a Work Around for the default openshift version not being available in NewClusterPage. If the openshift_versions were in a global context it wouldn't need to be updated in ClusterDetails at all.
The reason it is interactive, is that FeatureSupportLevelBadge rerender will have this updated value

Reply:
Why do we need to store the currently selected version in the FeatureSupportLevelData? If the version is selected, it is stored in the cluster. I feel like we're adding a lot of logic to manage the state of currently selected version inside the class instead of passing it as a parameter to the getters. If we removed that, we wouldn't need to do these effects and maybe event the class would not be necessary and we'd have just a set of functions for retrieving the support information provided by the context provider:

`getFeatureSupportLevel(featureId, openshiftVersion)`, `isVersionFullySupported(openshiftVersion)`, ...

Reply:
The idea of this implementation is that the badge is plug and play without properties of the cluster, so you don't need to drill down the cluster feature usage and cluster openshift version to every component that needs to know the support level of a specific version.
You are right that the drawback is that the openshift version is maintained in two places. 
This can be fixed if the cluster was stored in a global place and the FeatureSupportLevelProvider can access it. (It actually is currently stored in a Redux state but my understanding is that we don't want to use this)
In general this implementation isn't ideal because the data required for this context isn't available globally, so I needed to work around it to achieve the easy usage of the badge.
I'm considering refactoring this in a separate PR